### PR TITLE
ENG-281: Adds new section (DW disconnected)

### PR DESCRIPTION
**Pull request by bredamc**
First draft of the new section:
"Running distributed data science workloads in a disconnected environment"

### CODE REVIEW

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+:_module-type: PROCEDURE
+
+[id="running-distributed-data-science-workloads-disconnected-env_{context}"]
+= Running distributed data science workloads in a disconnected environment
+
+[role='_abstract']
```
Comment (chtyler) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:6]: Regarding this:
"..and you must set some environment variables"

As indicated in the procedure, these environment variables are being set to indicate the location of the Python dependencies.

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+. Run the distributed data science workload, as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#running-distributed-data-science-workloads-from-notebooks_distributed-workloads[Running distributed data science workloads from notebooks] or link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#running-distributed-data-science-workloads-from-ds-pipelines_distributed-workloads[Running distributed data science workloads from data science pipelines].
+
+.Verification
+The notebook or pipeline run completes without errors.
+
+////
+[role='_additional-resources']
+.Additional resources
+<Do we want to link to additional resources?>
+
```
Comment (chtyler) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:35]: Potential additional resources:

Ray cluster docs
Disconnected install docs

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
++
+----
+PIP_INDEX_URL: https://pypi-notebook.apps.mylocation.com/simple
+PIP_TRUSTED_HOST: pypi-notebook.apps.mylocation.com
+----
+. Edit the notebook or pipeline to provide the location of the Ray cluster.
+. Run the distributed data science workload, as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#running-distributed-data-science-workloads-from-notebooks_distributed-workloads[Running distributed data science workloads from notebooks] or link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#running-distributed-data-science-workloads-from-ds-pipelines_distributed-workloads[Running distributed data science workloads from data science pipelines].
+
+.Verification
+The notebook or pipeline run completes without errors.
```
Comment (chtyler) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:29]: Can the user also verify this from the user interface in the dashboard?

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+* You have access to the data sets and models to be used by the workload.
+* You have access to the Python dependencies for training scripts, either in a Ray image or in your own PyPi server that is available from the disconnected cluster.
+* You have created a data science project.
+
+.Procedure
+. Configure the disconnected data science cluster to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#configuring-distributed-workloads_distributed-workloads[Configuring distributed workloads].
+. Set the `PIP_INDEX_URL` and `PIP_TRUSTED_HOST` environment variables to indicate the location of the Python dependencies, as shown in the following example:
++
+----
+PIP_INDEX_URL: https://pypi-notebook.apps.mylocation.com/simple
```
Comment (chtyler) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:22]: Although it says that these indicate the location of the Python dependencies, it doesn't say what each environment variable actually is. Here is my guess from Googling them:

PIP_INDEX_URL: This is the base URL of the Python Package Index (default https://pypi.org/simple).
PIP_TRUSTED_HOST: Instructs Python to implicitly trust the specific host the package is downloaded from, whether or not it has a valid SSL certificate, or is using a secure channel at all.

**Comment by bredamc** [comment]
@chtyler Many thanks for your review comments, draft updated accordingly :)

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* You have installed {productname-long} and created a mirror image as described in link:{rhoaidocshome}{default-format-url}/installing_and_uninstalling_openshift_ai_self-managed_in_a_disconnected_environment[Installing and uninstalling {productname-short} Self-Managed in a disconnected environment].
+* You have access to a Ray cluster image.
+* You have access to the data sets and models to be used by the workload.
+* You have access to the Python dependencies for the workload, either in a Ray image or in your own Python Package Index (PyPI) server that is available from the disconnected cluster.
+* You have logged in to {productname-long}.
+* You have created a data science project.
+
+.Procedure
+. Configure the disconnected data science cluster to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#configuring-distributed-workloads_distributed-workloads[Configuring distributed workloads].
+. Edit the notebook or pipeline to provide the location of the Ray cluster image:
```
Comment (sutaakar) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:21]: Just a note - Ray cluster image has to be provided by user in all cases, even for "normal" deployment.
So, maybe something like "Make sure that Ray cluster image referenced in notebook or pipeline points to image available in disconnected environment."?

**Review by sutaakar — APPROVED** [review]

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+** The Python dependencies for the workload, either in a Ray image or in your own Python Package Index (PyPI) server that is available from the disconnected cluster
+* You have logged in to {productname-long}.
+* You have created a data science project.
+
+.Procedure
+. Configure the disconnected data science cluster to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#configuring-distributed-workloads_distributed-workloads[Configuring distributed workloads].
+. Ensure that the notebook or pipeline includes a link to a Ray cluster image that can be accessed from the disconnected environment:
+* Notebooks use the Ray cluster image to create a Ray cluster when running the notebook.
+* Pipelines use the Ray cluster image to create a Ray cluster during the pipeline run.
+. If any of the Python packages required by the workload are not available in the Ray cluster, configure the Ray cluster to download the Python packages from a private PyPI server.
```
Comment (dimakis) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:25]: Should we perhaps point the user to [this repo](https://github.com/project-codeflare/pypi-cache) which shows the install for a pypi server which we have tested works?

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+* You can access the following software from the disconnected cluster:
+** A Ray cluster image
+** The data sets and models to be used by the workload
+** The Python dependencies for the workload, either in a Ray image or in your own Python Package Index (PyPI) server that is available from the disconnected cluster
+* You have logged in to {productname-long}.
+* You have created a data science project.
+
+.Procedure
+. Configure the disconnected data science cluster to run distributed workloads as described in link:{rhoaidocshome}{default-format-url}/working_on_data_science_projects/working-with-distributed-workloads_distributed-workloads#configuring-distributed-workloads_distributed-workloads[Configuring distributed workloads].
+. Ensure that the notebook or pipeline includes a link to a Ray cluster image that can be accessed from the disconnected environment:
```
Comment (dimakis) [inline modules/running-distributed-data-science-workloads-disconnected-env.adoc:22]:
```suggestion
. During the instantiation of a Ray Cluster via the `ClusterConfiguration` code, ensure that the notebook or pipeline includes a link to a Ray cluster image that can be accessed from the disconnected environment:
```

Here is an [example of exactly](https://github.com/project-codeflare/codeflare-sdk/blob/62be50e23e7ecba074e540c4715f0dd9f4ca33a0/demo-notebooks/guided-demos/2_basic_jobs.ipynb#L59-L70) what this is along with a community demo image on line 69 which is the image we use for our internal testing hosted in our own public registry.

File: modules/running-distributed-data-science-workloads-disconnected-env.adoc (hunk 6 above)
Comment (dimakis) [reply modules/running-distributed-data-science-workloads-disconnected-env.adoc:25]: not needed

**Review by dimakis — APPROVED** [review]
lgtm, great work as always

**Review by chtyler — APPROVED** [review]

**Pull request merged**